packages/resilience/src/timeout.spec.ts

```diff
@@ -0,0 +1,59 @@
+import nock from 'nock';
+import { executeHttpRequest } from '@sap-cloud-sdk/http-client';
+import type { Destination } from '@sap-cloud-sdk/connectivity';
+import { timeout } from './timeout';
+
+describe('timeout', () => {
+  const httpsDestination: Destination = {
+    name: 'httpsDestination',
+    url: 'https://example.com'
+  };
+
+  it('considers timeout via middleware on csrf token fetching', async () => {
+    const delayInResponse = 10;
+    nock('http://example.com', {})
+      .post(/with-delay/)
+      .delay(delayInResponse)
+      .reply(200);
+
+    await expect(
+      executeHttpRequest(
+        { url: 'http://example.com' },
+        {
+          method: 'post',
+          url: 'with-delay',
+          middleware: [timeout(delayInResponse * 0.5)]
```

Comment:
[pp] I would switch one of the 2 tests to `timeout()` for testing default value, or add a new test